### PR TITLE
Display banner if maintenance is scheduled

### PR DESCRIPTION
**Pull request by VKTB**
## Description
Fetches the scheduled maintenance state from `scigateway-auth` and loads it into the store, If the `show` boolean flag is `True`, it 
displays a warning.

I'd like to get opinions on whether the logic for displaying the banner should live in `scigateway.actions.tsx ` or somewhere else.

## Testing instructions
You will either need to pull the changes from the [Implement maintenance and scheduled maintenance endpoints](https://github.com/ral-facilities/scigateway-auth/pull/52) PR and deploy SciGateway-Auth locally, or change the initial state of [scheduledMaintenance](https://github.com/ral-facilities/scigateway/commit/8c9f5b9bf403cee9fa7244f1e75603211ac43139#diff-0059ae3dab3662679f6946ca06529176f5e59697aef897b75a44434faf10a53cR66) in the reducer.

- [x] Review code
- [x] Check Actions build
- [x] Review changes to test coverage
- [x] No banner is displayed when the `show` boolean flag returned by `scigateway-auth` is `False`.
- [x] Warning is displayed when the `show` boolean flag returned by `scigateway-auth` is `True`. Check that the warning message is the same as what's returned by the backend.

## Agile board tracking
connect to #514

### CODE REVIEW

File: src/authentication/icatAuthProvider.tsx [hunk 1]
```diff
@@ -85,4 +86,14 @@ export default class ICATAuthProvider extends BaseAuthProvider {
         this.handleRefreshError(err);
       });
   }
+
+  public fetchScheduledMaintenanceState(): Promise<ScheduledMaintenanceState> {
+    return Axios.get(`${this.authUrl}/scheduled_maintenance`)
+      .then((res) => {
+        return res.data;
+      })
+      .catch((err) => {
```
Comment (patrick-austin) [inline src/authentication/icatAuthProvider.tsx:95]: Not covered by tests.

File: src/state/actions/scigateway.actions.tsx [hunk 2]
```diff
@@ -245,9 +258,38 @@ export const configureSite = (): ThunkResult<Promise<void>> => {
...
+    if (provider.fetchScheduledMaintenanceState) {
+      provider
+        .fetchScheduledMaintenanceState()
+        .then((scheduledMaintenaceState) => {
+          dispatch(loadScheduledMaintenanceState(scheduledMaintenaceState));
+
+          const scheduledMaintenance = getState().scigateway
+            .scheduledMaintenance;
+          if (scheduledMaintenance['show']) {
+            displayScheduledMaintenanceBanner(scheduledMaintenance['message']);
```
Comment (patrick-austin) [inline src/state/actions/scigateway.actions.tsx:272]: Not covered by tests.

**Review by patrick-austin — DISMISSED** [review]
Works fine. Test coverage is high enough to merge as is, but could be improved by adding one for error handling and checking that loading a payload with `show: true` so that `displayScheduledMaintenanceBanner` is called.

**Comment by VKTB** [comment]
@patrick-austin Added new tests to increase the coverage but also note the changes that I made in commit [6164ce8](https://github.com/ral-facilities/scigateway/pull/554/commits/6164ce8f06f81104fdc5cd62baedfec3a2d21526) - The scheduled maintenance state was not getting loaded into the store before the conditional statement so if the backend was returning `show: true` then the logic inside the conditional statement was not getting executed because `show` is still `false` as that is what the initial state is set to be.

**Review by patrick-austin — APPROVED** [review]

**Pull request merged**